src/SDKs/ApiManagement/ApiManagement.Tests/ApiManagementTestBase.cs

```diff
@@ -0,0 +1,166 @@
+            this.resourcesClient = context.GetServiceClient<ResourceManagementClient>();
+            this.storageClient = context.GetServiceClient<StorageManagementClient>();
+            this.networkClient = context.GetServiceClient<NetworkManagementClient>();
+            this.eventHubClient = context.GetServiceClient<EventHubManagementClient>();
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            var testEnv = TestEnvironmentFactory.GetTestEnvironment();
+
+            if (HttpMockServer.Mode == HttpRecorderMode.Record)
+            {
+                this.serviceName = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(ServiceNameKey);
+                if (string.IsNullOrEmpty(this.serviceName))
+                {
+                    this.serviceName = TestUtilities.GenerateName("sdktestapim");
+                }
+
+                this.location = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(LocationKey);
+                if (string.IsNullOrEmpty(this.location))
+                {
+                    this.location = GetLocation();
+                }
+
+                this.rgName = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(ResourceGroupNameKey);
+                if (string.IsNullOrEmpty(this.rgName))
+                {
+                    rgName = TestUtilities.GenerateName("sdktestrg");
+                    resourcesClient.ResourceGroups.CreateOrUpdate(rgName, new ResourceGroup { Location = this.location });
+                }
+                                
+                this.base64EncodedTestCertificateData = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(TestCertificateKey);
+                if (!string.IsNullOrEmpty(this.base64EncodedTestCertificateData))
+                {
+                    HttpMockServer.Variables[TestCertificateKey] = base64EncodedTestCertificateData;
+                }
+                this.testCertificatePassword = testEnv.ConnectionString.KeyValuePairs.GetValueUsingCaseInsensitiveKey(TestCertificatePasswordKey);
+                if(!string.IsNullOrEmpty(this.testCertificatePassword))
+                {
+                    HttpMockServer.Variables[TestCertificatePasswordKey] = testCertificatePassword;
+                }
+
+                this.tenantId = testEnv.Tenant;
+                this.subscriptionId = testEnv.SubscriptionId;
+                HttpMockServer.Variables[TenantIdKey] = tenantId;
+                HttpMockServer.Variables[SubIdKey] = subscriptionId;                
+                HttpMockServer.Variables[ServiceNameKey] = this.serviceName;
+                HttpMockServer.Variables[LocationKey] = this.location;
+                HttpMockServer.Variables[ResourceGroupNameKey] = this.rgName;
+            }
+            else if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                this.tenantId = testEnv.Tenant;
```

Comment:
remove some extra variables. I require some initialization during playback, as some of the tests can only be executed on an active service. Service Activation takes 40 minutes, hence use this as a place to provide active service details, and proceed with additional validation.